vmdb/app/models/miq_provision/naming.rb

```diff
@@ -51,28 +52,20 @@ def get_vm_full_name(unresolved_vm_name, prov_obj, determine_index)
       # if we are just building a sample of what the vm_name will look like use '#' inplace of actual number.
       return "#{name[:prefix]}#{'#' * name[:index_length]}#{name[:suffix]}" if determine_index == false
 
-      # Determine starting index based on already assigned names in miq_provision table.
-      start_idx = 0
-      reg_val = Regexp.new("(#{name[:prefix]})(\\d{#{name[:index_length]}})(#{name[:suffix]})")
-
-      MiqProvision.find(:all).each do |p|
-        if p.options[:vm_target_name].to_s.strip =~ reg_val
-          name_prefix, name_idx, name_suffix = $1.to_s, $2.to_i, $3
-          # If the name prefix and suffix match record the highest index number
-          if name[:prefix].downcase == name_prefix.downcase && name[:suffix].to_s.downcase == name_suffix.to_s.downcase
-            start_idx = name_idx if name_idx > start_idx
-          end
+      index_length = name[:index_length]
+      loop do
+        next_number = MiqRegion.my_region.next_naming_sequence(unresolved_vm_name, SOURCE_IDENTIFIER)
+        idx_str  = "%0#{index_length}d" % next_number
```

Comment:
@kbrock Good to learn this way. I will leave as is for now since there are no other changes needed for this PR.